### ISSUE TITLE
Duo strength version 2.0.12 stopped working 

### ISSUE THREAD

**Issue by jim-cress** [ToranSharma/Duo-Strength#182]
this week when duolingo updated to allow users to see legendary (purple) skills which were first introduced on the android platform. As soon as Duolingo updated the duo strength stopped working. see pic
![image](https://user-images.githubusercontent.com/78605026/133503040-57108da0-3645-4c73-9432-35aa26be6a97.png)
notice at the top the duo strength is green indicating it is running, i have been through the menus all the side boxes are turned on. I have tried uninstalling and reinstalling duo strength, to no avail. Are you aware of this problem?

Jim Cress
1-705-280-9076

**Comment by ToranSharma** [comment]
Hi again Jim,

Thanks for letting me know about the issue. Another user has also reported the same thing with the new purple legendary skills.

Unfortunately this feature has not been activated for all Duolingo users yet, and none of the accounts that I have access to have it. As such I cannot start fixing the issue until I get to see it myself.

I will keep checking each day to see if I can see and then fix the issue.

Thanks for your support, sorry I can't get this sorted as quickly I would like. I let you know as soon as I have an update.

Toran

**Comment by jim-cress** [comment]
No thats awesome, thank you. I have been playing around, i can get the
strength bars working but not the side boxes or mastered buttons.... Im
sure that Duo broke something! I appreciate you extension so much. Have a
great day.

Jim

**Comment by jim-cress** [comment]
Any word on this? Did you get an update on your dyolingo?

Jim

**Comment by ToranSharma** [comment]
Hi Jim,

Still no update on any of the accounts I have access to. I will see if I can force it to happen by using the android app which will have the legendary skills feature. It is possible that duolingo is focusing on rolling out the update to the website to people who use the app.

I'll keep you posted with any updates. Sorry again for the delay in getting this sorted.

Toran

**Comment by jim-cress** [comment]
Absolutely no worries, i use the android app and microsoft edge on my pc. I
absolutely love your extension! Have a great day.

**Comment by ToranSharma** [comment]
Scratch that, I have just logged out an logged in again and the switch has happened! The strange thing is that most the main features are still working for me. It may be that I need to have at least one legendary skill for it to break.

I'll try to sort out all the issues as soon as I can.

**Comment by jim-cress** [comment]
I can get some working sometimes, then i do a lesson and it doesnt reload
the extension. Also the crowns info doesnt work.

**Comment by ToranSharma** [comment]
Hi Jim,

I have just release v2.0.13 which should be pushed out to browsers in the next couple of hours. This update hopefully fixes all the issues that I could see regarding the new legendary skills.

If you have any further issues once you have received the update please let me know.

Toran

PS Here is the list of changes in this update:

### Fixed
- Issues with new legendary skills.
  - Skill list links point to practice if skill is legendary.
    - L5 normal skills, L2 grammar skills and L1 bonus skill links point to the legendary test.
  - Practise button adding handles new crown level option.
  - Tips page practice button adding.
  - Maximum crown count calculation updated to account for legendary skills.
  - Crowns breakdown table now includes legendary level.
  - Test out button being added to L2 grammar skills.

### Added
- New flag border colour and legendary crown logo for L6 trees.
- Option to add a legendary button to the top the tips page of skills of the
  correct level.
  - Enabled by default.
- Maximum value for lesson threshold option to 5.
  - With this option set to 5, L5 skill links in at the top of the tree will practice and not point to the legendary test (the default behaviour).

**Comment by jim-cress** [comment]
This is Awesome! checking it out now, looks amazing and the purple
legendary stuff work!

jim

**Comment by jim-cress** [comment]
I have 2 items, 1 the lessons toward next level in the crowns box does not
see to be calculating, see screen shotbelow.
2 you can remove legendary cost! WOW AWESOME!
[image: image.png]

**Comment by ToranSharma** [comment]
Hi Jim,

Unfortunately your image didn't attach properly, I think you need to do it from the github site itself, not in an email.

Regarding the removal of the legendary cost, I assume you mean that clicking a link doesn't take you to the lingot "payment" screen, just straight into the test. I thought this too so I did test this and if I remember correctly the 10 lingot cost was just taken after the test was complete. I had made an option to replace the legendary test button with a version that skips this screen if the cost really isn't taken, but on finding it was I removed it.

Toran

**Comment by jim-cress** [comment]
Yes that is awesome! The pic shoew the # of lessons toward next level for
the 7 days

**Comment by ToranSharma** [comment]
Oh I see do you mean the graph, that is added above the crowns breakdown table? Something like this?

![image](https://user-images.githubusercontent.com/4559968/134819320-2c0df675-9d2a-4336-9924-bc4898f8819c.png)

It seems to be working for me, have just done a lesson and you can see that today has gone up to 1.

**Comment by jim-cress** [comment]
Thats the one, mine doesnt seem to work, not major. The legendary being
free more than compensates!

**Comment by ToranSharma** [comment]
I'll have a look into it a bit more to see If I can see any issues with it. Note that it is not a graph of crowns earned, it is a graph of the number of lessons you have done which were for skills that are at the lowest crown level.

For example, if you have yet to finish a tree, the lowest level of skills you will have are L0. Doing a lesson a new skill will contribute to your progress towards finishing the tree, and thus show up on the graph. Doing a lesson on a skill you have already completed, i.e. L1 or higher, doesn't contribute towards you finishing the tree and so is not counted on the graph.

Also sorry if I was a bit unclear in my previous message, but as far as I know, the legendary is NOT free, you are still charged the 10 lingots, just not shown the screen before hand. I could be wrong and will try to test it again, let me know if your experience is different.

Toran

**Comment by jim-cress** [comment]
now its clear, also yes i had to pay...

Still appreciate all your hard work!

jim

**Comment by jim-cress** [comment]
I think i found the problem, the crowns sidebar at the bottom where it
calculates the number of days to your next skill. I have completed level 3
and am presently doing level 4 skills however my calculator tells me 20
days until level 3, i have 100% strength and only 3 cracked skills. on the
web version of duolingo you can not see the  level crowns for OWl
statue but on android app you can. On android i have 3 crowns below statue.
this is probably why the level indicator isn't working, because i am not
working  level 3 skills. Therefore it wouldn't register, however the level
indicator isn't working.

**Comment by jim-cress** [comment]
submitted new issue on Github, with 2 screenshots for clarification.

jim

**Comment by ToranSharma** [comment]
Thanks Jim, I have replied there.
